### ISSUE TITLE
curly multi-or-nest incorrect example with a comment

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12972]
**Tell us about your environment**

* **ESLint Version:** v7.0.0-alpha.1
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015,
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgY3VybHk6IFtcImVycm9yXCIsIFwibXVsdGktb3ItbmVzdFwiXSovXG5cbmlmIChmb28pXG4gICAgLy8gc29tZSBjb21tZW50XG4gICAgYmFyKCk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) (v6.8.0)

```js
/*eslint curly: ["error", "multi-or-nest"]*/

if (foo)
    // some comment
    bar();
```

```bash
eslint index.js
```

**What did you expect to happen?**

In [curly multi-or-nest documentation](https://eslint.org/docs/rules/curly#multi-or-nest) this is one of the examples of **incorrect** code (the last one).

**What actually happened? Please include the actual, raw output from ESLint.**

no errors.

**Are you willing to submit a pull request to fix this bug?**

I'm not sure should this be fixed in the rule or in the docs. This may be a bug, but I have concerns about the impact if we change the behavior now.

Relevant issue/PRs: #7538, #7539, #7597.

The intention was to allow this:

```js
/*eslint curly: ["error", "multi-or-nest"]*/

if (foo) {
    // some comment
    bar();
}
```

but I can't tell from the relevant discussions whether the intention was also to disallow the code that was added to the incorrect examples.

**Comment by kaicataldo** [comment]
Nice find! From the documentation:

> You can use another configuration that forces brace-less if, else if, else, for, while, or do if their body contains only one single-line statement.

Given this heuristic, I don't think this should warn. That being said, it looks like the reason this isn't warning is because it ignores cases where there are comments (which I'm not sure is expected behavior) (see demo [here](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgY3VybHk6IFtcImVycm9yXCIsIFwibXVsdGktb3ItbmVzdFwiXSovXG5cbmlmIChmb28pIC8vIGEgcHJldmlvdXMgY29tbWVudFxuICAgIC8vIHNvbWUgY29tbWVudFxuICAgLy8gYW5vdGhlciBjb21tZW50XG4gICAgYmFyKCk7XG5cblxuaWYgKGZvbylcbiAgICBiYXIoKTtcblxuaWYgKGZvbykgey8vIGEgcHJldmlvdXMgY29tbWVudFxuICAgIC8vIHNvbWUgY29tbWVudFxuICAgLy8gYW5vdGhlciBjb21tZW50XG4gICAgYmFyKCk7XG59XG5cbmlmIChmb28pIHtcbiAgICBiYXIoKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)).

**Comment by kaicataldo** [comment]
I would advocate for updating the documentation to match the behavior of the rule right now and to follow up with a separate PR if we decide that the behavior is incorrect.

**Comment by mdjermanovic** [comment]
PR #13151 updates the docs to match the actual behavior.

I'd vote to leave the rule as is. This could be unintended behavior, but fixing it looks like a big change for users now.
